sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -20,12 +20,17 @@
 
 #include "net/gnrc/ipv6/ext.h"
 
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
 bool gnrc_ipv6_ext_demux(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
                          uint8_t nh)
 {
-    gnrc_pktsnip_t *ext_snip;
+    gnrc_pktsnip_t *ext_snip, *tmp;
     ipv6_ext_t *ext;
     unsigned int offset = 0;
+    ipv6_hdr_t *hdr;
+    int res;
 
     ext = ((ipv6_ext_t *)(((uint8_t *)pkt->data) + sizeof(ipv6_hdr_t)));
```

Comment:
I don't understand why we need to update `ext` before the call to `ipv6_ext_rh_process`?
`ext` will be updated for each header extension (that is not a routing header extension) that was encountered before. Once we reach the routing header extension, `ext` should point to the right place? That's why `nh` is updated before getting the next `ext` in line ~79.